npm/controlplane.go

```diff
@@ -0,0 +1,144 @@
+package npm
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	npmconfig "github.com/Azure/azure-container-networking/npm/config"
+	controllersv2 "github.com/Azure/azure-container-networking/npm/pkg/controlplane/controllers/v2"
+	"github.com/Azure/azure-container-networking/npm/pkg/dataplane"
+	"github.com/Azure/azure-container-networking/npm/pkg/transport"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/version"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
+)
+
+type NetworkPolicyControlplane struct {
+	config npmconfig.Config
+
+	// tm is the transport layer (gRPC) manager/server
+	tm *transport.Manager
+
+	// Informers are the Kubernetes Informer
+	// https://pkg.go.dev/k8s.io/client-go/informers
+	Informers
+
+	// Controllers for handling Kubernetes resource watcher events
+	K8SControllersV2
+
+	// Azure-specific variables
+	AzureConfig
+}
+
+func NewNetworkPolicyControlplane(
+	config npmconfig.Config,
+	informerFactory informers.SharedInformerFactory,
+	dp dataplane.GenericDataplane,
+	npmVersion string,
+	k8sServerVersion *version.Info,
+) *NetworkPolicyControlplane {
+	klog.Infof("API server version: %+v AI metadata %+v", k8sServerVersion, aiMetadata)
+
+	mgr := transport.NewManager(context.Background(), config.Transport.Port)
+
+	// TODO (nitishm): This needs to be passed to the Dataplane instance
+	_ = mgr.InputChannel()
```

Comment:
i think we will need to push TM one step above, because, DP expects the input channel to be provided when the object is created. So start.go (for controller) can be our main orch point, no strong opinion though